tests/unit_tests/epee_utils.cpp

```diff
@@ -1751,65 +1751,89 @@ TEST(parsing, number)
   boost::string_ref val;
   std::string s;
   std::string::const_iterator i;
+  bool is_float_val;
+  bool is_signed_val;
 
   // the parser expects another character to end the number, and accepts things
   // that aren't numbers, as it's meant as a pre-filter for strto* functions,
   // so we just check that numbers get accepted, but don't test non numbers
 
   s = "0 ";
   i = s.begin();
-  epee::misc_utils::parse::match_number(i, s.end(), val);
-  ASSERT_EQ(val, "0");
+  epee::misc_utils::parse::match_number2(i, s.end(), val, is_float_val, is_signed_val);
+  EXPECT_EQ(val, "0");
+  EXPECT_FALSE(is_float_val);
+  EXPECT_FALSE(is_signed_val);
 
   s = "000 ";
   i = s.begin();
-  epee::misc_utils::parse::match_number(i, s.end(), val);
-  ASSERT_EQ(val, "000");
+  epee::misc_utils::parse::match_number2(i, s.end(), val, is_float_val, is_signed_val);
+  EXPECT_EQ(val, "000");
+  EXPECT_FALSE(is_float_val);
+  EXPECT_FALSE(is_signed_val);
 
   s = "10x";
   i = s.begin();
-  epee::misc_utils::parse::match_number(i, s.end(), val);
-  ASSERT_EQ(val, "10");
+  epee::misc_utils::parse::match_number2(i, s.end(), val, is_float_val, is_signed_val);
+  EXPECT_EQ(val, "10");
+  EXPECT_FALSE(is_float_val);
+  EXPECT_FALSE(is_signed_val);
 
   s = "10.09/";
   i = s.begin();
-  epee::misc_utils::parse::match_number(i, s.end(), val);
-  ASSERT_EQ(val, "10.09");
+  epee::misc_utils::parse::match_number2(i, s.end(), val, is_float_val, is_signed_val);
```

Comment:
and here obviously:

is_float_val = false; is_signed_val = true`

and so on...